File: src/models/battery_data.rs
use upower_dbus::BatteryState;

#[derive(Clone, PartialEq, Debug)]
pub struct BatteryData {
    pub percentage: f64,
    pub state: BatteryState,
    pub health: f64,
    pub temperature: f64,
    pub rate: f64,
    pub time_to_empty: i64,
    pub time_to_full: i64,
    pub serial: String,
    pub vendor: String,
    pub model: String,
    pub voltage: f64,
    pub charge_cycles: i64,
    pub history_percentage: Vec<(u32, f64)>,
    pub history_rate: Vec<(u32, f64)>,
}

File: src/types/battery_data.ts
export enum BatteryState {
	Unknown = 0,
	Charging = 1,
	Discharging = 2,
	Empty = 3,
	FullyCharged = 4,
	PendingCharge = 5,
	PendingDischarge = 6,
}

export type BatteryStateType =
	| "Unknown"
	| "Charging"
	| "Discharging"
	| "Empty"
	| "Fully Charged"
	| "Pending Charge"
	| "Pending Discharge"

export interface BatteryData {
	percentage: number
	state: BatteryState
	health: number
	temperature: number
	rate: number
	timeToEmpty: number
	timeToFull: number
	serial: string
	vendor: string
	model: string
	voltage: number
	chargeCycles: number
	historyPercentage: Array<[number, number]>
	historyRate: Array<[number, number]>
}
